scipy/interpolate/tests/test_bary_rational.py

```diff
@@ -235,3 +258,95 @@ def test_spiral_cleanup(self):
+            FloaterHormannInterpolator([np.inf], [1], d=0)
+        with pytest.raises(ValueError, match="`d`"):
+            FloaterHormannInterpolator([0], [0], d=-1)
+        with pytest.raises(ValueError, match="`d`"):
+            FloaterHormannInterpolator([0], [0], d=10)
+        with pytest.raises(TypeError):
+            FloaterHormannInterpolator([0], [0], d=0.0)
+
+    # reference values from Floater and Hormann 2007 page 8.
+    @pytest.mark.parametrize("d,expected", [
+        (0, [1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1]),
+        (1, [1, 2, 2, 2, 2, 2, 2, 2, 2, 2, 1]),
+        (2, [1, 3, 4, 4, 4, 4, 4, 4, 4, 3, 1]),
+        (3, [1, 4, 7, 8, 8, 8, 8, 8, 7, 4, 1]),
+        (4, [1, 5, 11, 15, 16, 16, 16, 15, 11, 5, 1])
+    ])
+    def test_uniform_grid(self, d, expected):
+        # Check against explicit results on an uniform grid
+        x = np.arange(11)
+        r = FloaterHormannInterpolator(x, 0.0*x, d=d)
+        assert_allclose(r.weights.ravel()*self.scale(x.size, d), expected,
+                        rtol=1e-15, atol=1e-15)
+
+    @pytest.mark.parametrize("d", range(10))
+    def test_runge(self, d):
+        x = np.linspace(0, 1, 51)
+        rng = np.random.default_rng(802754237598370893)
+        xx = rng.uniform(0, 1, size=1000)
+        y = self.runge(x)
+        h = x[1] - x[0]
+
+        r = FloaterHormannInterpolator(x, y, d=d)
+
+        tol = 10*h**(d+1)
+        assert_allclose(r(xx), self.runge(xx), atol=1e-10, rtol=tol)
+        # check interpolation property
+        assert_equal(r(x), self.runge(x))
+
+    def test_complex(self):
+        x = np.linspace(-1, 1)
+        z = x + x*1j
+        r = FloaterHormannInterpolator(z, np.sin(z), d=12)
+        xx = np.linspace(-1, 1, num=1000)
+        zz = xx + xx*1j
+        assert_allclose(r(zz), np.sin(zz), rtol=1e-12)
+
+    def test_polyinterp(self):
+        # check that when d=n-1 FH gives a polynomial interpolant
+        x = np.linspace(0, 1, 11)
+        xx = np.linspace(0, 1, 1001)
+        y = np.sin(x)
+        r = FloaterHormannInterpolator(x, y, d=x.size-1)
+        p = BarycentricInterpolator(x, y)
+        assert_allclose(r(xx), p(xx), rtol=1e-12, atol=1e-12)
+
+    @pytest.mark.parametrize("y_shape", [(2,), (2, 3, 1), (1, 5, 6, 4)])
+    @pytest.mark.parametrize("xx_shape", [(100), (10, 10)])
+    def test_trailing_dim(self, y_shape, xx_shape):
+        x = np.linspace(0, 1)
+        y = np.broadcast_to(
+            np.expand_dims(np.sin(x), tuple(range(1, len(y_shape) + 1))),
+            x.shape + y_shape
+        )
+
+        r = FloaterHormannInterpolator(x, y)
+
+        rng = np.random.default_rng(897138947238097528091759187597)
+        xx = rng.random(xx_shape)
+        yy = np.broadcast_to(
+            np.expand_dims(np.sin(xx), tuple(range(xx.ndim, len(y_shape) + xx.ndim))),
+            xx.shape + y_shape
+        )
+        rr = r(xx)
+        assert rr.shape == xx.shape + y_shape
+        assert_allclose(rr, yy, rtol=1e-6)
```

Comment:
Any specific reason this tolerance this loose? If it's an interpolator, I'd normally expect something closer to `atol=1e-15` maybe with a bump of a couple orders of magnitude.

Reply:
That seems quite tight for randomly chosen points?

Reply:
So it's from what you actually get running the test locally? OK then, nevermind.